src/components/scenes/MigrateWalletCalculateFeeScene.tsx

```diff
@@ -0,0 +1,268 @@
+    const fee = feeState[key]
+
+    let rightSide: JSX.Element
+    if (fee == null) {
+      rightSide = <ActivityIndicator style={{ paddingRight: theme.rem(0.3125) }} color={theme.iconTappable} />
+    } else if (fee instanceof Error) {
+      rightSide = <EdgeText style={{ color: theme.negativeText, fontSize: theme.rem(0.75) }}>{fee.message}</EdgeText>
+
+      if (fee instanceof InsufficientFundsError) {
+        return (
+          <CreateWalletSelectCryptoRow
+            pluginId={pluginId}
+            tokenId={tokenId}
+            walletName={walletName}
+            rightSide={rightSide}
+            onPress={() => {
+              handleInsufficientFunds(wallet, fee)
+            }}
+          />
+        )
+      }
+    } else {
+      const fakeEdgeTransaction = {
+        currencyCode,
+        nativeAmount: '0',
+        networkFee: fee,
+        blockHeight: 0,
+        date: 0,
+        txid: '',
+        signedTx: '',
+        ourReceiveAddresses: [],
+        walletId
+      }
+      const exchangeDenom = denominations.find(denom => denom.name === currencyCode) as EdgeDenomination
+      const displayDenom = displayDenominations[pluginId]?.[currencyCode] ?? exchangeDenom
+
+      const transactionFee = convertTransactionFeeToDisplayFee(wallet, exchangeRates, fakeEdgeTransaction, displayDenom, exchangeDenom)
+      const fiatAmount = transactionFee.fiatAmount === '0' ? '0' : ` ${transactionFee.fiatAmount}`
+      const feeSyntax = `${transactionFee.cryptoSymbol ?? ''} ${truncateDecimals(transactionFee.cryptoAmount)} (${
+        transactionFee.fiatSymbol ?? ''
+      }${fiatAmount})`
+      rightSide = <EdgeText style={{ color: theme.secondaryText, fontSize: theme.rem(0.75) }}>{feeSyntax}</EdgeText>
+    }
+
+    return <CreateWalletSelectCryptoRow pluginId={pluginId} tokenId={tokenId} walletName={walletName} rightSide={rightSide} />
+  })
+
+  const handleInsufficientFunds = useHandler(async (wallet, error) => {
+    await Airship.show(bridge => <InsufficientFeesModal bridge={bridge} coreError={error} navigation={navigation} wallet={wallet} />)
+  })
+
+  const handleSlidingComplete = useHandler(() => {
+    const filteredMigrateWalletList = migrateWalletList.filter(asset => typeof feeState[asset.key] === 'string')
+    navigation.push('migrateWalletCompletion', { migrateWalletList: filteredMigrateWalletList })
+  })
+
+  // Create getMaxSpendable/makeSpend promises for each selected asset. We'll group them by wallet first and then execute all of them while keeping
+  // track of which makeSpends are successful so we can enable the slider. A single failure from any of a wallet's assets will cast them all as failures.
+  useAsyncEffect(async () => {
+    // This bundles the assets by similar walletId with the main asset (ie. ETH) at the end of each array so its makeSpend is called last
+    const bundledWalletAssets: MigrateWalletItem[][] = migrateWalletList.reduce((bundles: MigrateWalletItem[][], asset) => {
```

Comment:
This one isn't actually O(n^2), because of a subtle difference in the way you have coded it. You could just as easily use a `for of` loop here, but it's not as important as the other places.